### ISSUE TITLE
Pass keyword arguments in keras CTC to tensorflow CTC function

### ISSUE THREAD

**Issue by brilliant-ember** [tensorflow/tensorflow#58775]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

2.8

### Custom Code

Yes

### Python version

3.10

### Current Behaviour?

```shell
Running a CTC-loss training can cause an error if the label is larger than the model's output. This error can be converted to a warning if you pass the approperiate flag. However, using keras ctc function (which is a wrapper to the tensorflow ctc function) doesn't allow the passing of key-word arguments to the ctc function in tensorflow.

The error text is 

Not enough time for target transition sequence (required: 156, available: 13)0You can turn this error into a warning by using the flag ignore_longer_outputs_than_inputs
```
```


### Standalone code to reproduce the issue

```shell
The code in this keras tutorial https://keras.io/examples/audio/ctc_asr/
```

### Relevant log output

```shell
File "/home/username/.local/lib/python3.10/site-packages/keras/engine/training.py", line 1082, in compute_loss
      return self.compiled_loss(
    File "/home/username/.local/lib/python3.10/site-packages/keras/engine/compile_utils.py", line 265, in __call__
      loss_value = loss_obj(y_t, y_p, sample_weight=sw)
    File "/home/username/.local/lib/python3.10/site-packages/keras/losses.py", line 152, in __call__
      losses = call_fn(y_true, y_pred)
    File "/home/username/.local/lib/python3.10/site-packages/keras/losses.py", line 284, in call
      return ag_fn(y_true, y_pred, **self._fn_kwargs)
    File "/tmp/ipykernel_8261/142572972.py", line 10, in CTCLoss
      loss = keras.backend.ctc_batch_cost(y_true, y_pred, input_length, label_length)
    File "/home/username/.local/lib/python3.10/site-packages/keras/backend.py", line 7023, in ctc_batch_cost
      tf.compat.v1.nn.ctc_loss(
Node: 'CTCLoss/CTCLoss'
2 root error(s) found.
  (0) INVALID_ARGUMENT:  Not enough time for target transition sequence (required: 156, available: 13)0You can turn this error into a warning by using the flag ignore_longer_outputs_than_inputs
	 [[{{node CTCLoss/CTCLoss}}]]
	 [[CTCLoss/CTCLoss/_86]]
  (1) INVALID_ARGUMENT:  Not enough time for target transition sequence (required: 156, available: 13)0You can turn this error into a warning by using the flag ignore_longer_outputs_than_inputs
	 [[{{node CTCLoss/CTCLoss}}]]
0 successful operations.
0 derived errors ignored. [Op:__inference_train_function_4574]
```
</details>

**Comment by brilliant-ember** [comment]
suggested edit is to the file `python/keras/backend.py` , just adding **kwargs passing to the function should in theory allow the passing of the flag to the ctc function in tensorflow
![image](https://user-images.githubusercontent.com/23269759/205464538-ed6fc5a4-b917-4ef9-abb6-7132c935e277.png)

**Comment by brilliant-ember** [comment]
### Suggested workaround 

in `/python/ops/ctc_ops.py` update the key-word argument flag `ignore_longer_outputs_than_inputs` to True 

![image](https://user-images.githubusercontent.com/23269759/205464758-dd12fc57-6931-489a-81bc-89d79d8d6d2d.png)

**Comment by b-bhupendra** [comment]
Hey, can I take this issue ?

**Comment by SuryanarayanaY** [comment]
Hi @brilliant-ember ,

Could you please confirm whether you have done any changes to the code in documentation that causing the error mentioned? Could you please share us a use case of the requirement if any. I tested the same code in documentation without any error and the [gist](https://colab.research.google.com/gist/SuryanarayanaY/917a1eafb29358a7518587219296722e/58775.ipynb) attached here.

Could you please confirm whether you have tested this code with changes you suggested?Was it worked for you?

Thankyou.

**Comment by brilliant-ember** [comment]
The suggested workaround worked for me, but the suggested fix didn't. because the my runtime wouldn't register my code change even if I restart my python runtime. The suggested workaround worked and registered the code change after a runtime restart. So updating the flag in `/python/ops/ctc_ops.py` worked but updated in `python/keras/backend.py` didn't even register and the runtime error didn't show my changes, maybe it has to do with the annotations, or my env isn't set up right.
To reproduce the error you need less time frames in the model, so increasing the time window and the frame step should do the trick.  I changed the vars to the ones below  in my local machine and the error occurred, I am running the collab now to confirm the issue happens there too.

``` python
frame_length = 1024
# An integer scalar Tensor. The number of samples to step.
frame_step = 512
```

**Comment by brilliant-ember** [comment]
In general this exception is raised when there's less timeframes than label length. When it is raised that means that the ctc model is not training for that sample. But sometimes you get cases where the label is just too short so it stops the training on an exception while it could have been just a small warning.

**Comment by sachinprasadhs** [comment]
Hi, Thanks for opening the issue. 
Usually,` keras/backend.py `uses legacy code, in most cases it uses `compat.v1` mode to ensure the backward compatibility. 
It is suggested to use native tensorflow API for any specific use case. Thanks!

**Comment by sachinprasadhs** [comment]
Existing keras implementation https://github.com/keras-team/keras/blob/4bc1eadedf19543b1445cb4b95e6f5502ef4ed82/keras/backend.py#L7019-L7022

Tensorflow implementation:
https://github.com/tensorflow/tensorflow/blob/941d9e5c8e9479bd666715ab6a5c2ce311141edf/tensorflow/python/ops/ctc_ops.py#L69-L78
